dlrover/python/master/watcher/k8s_watcher.py

```diff
@@ -134,7 +134,12 @@ def _convert_pod_yaml_to_node(pod):
     pod_id = int(metadata.labels[replica_index_key])
     rank_id = int(metadata.labels[rank_index_key])
     relaunch_count = int(metadata.labels[relaunch_count_key])
-    resource = _parse_container_resource(pod.spec.containers[0])
+    resource = NodeResource(0, 0)
+    for container in pod.spec.containers:
+        res = _parse_container_resource(container)
+        if res.cpu > 0 and res.memory > 0:
+            resource = res
```

Comment:
It is not appropriate to directly judge if the value is >0. 

Since a node is equivalent to a pod and this resource field is used to create a pod, I think it is acceptable to directly take the maximum value. Additionally, the GPU attribute should be included.